src/autolabel/few_shot/label_diversity_example_selector.py

```diff
@@ -0,0 +1,153 @@
+"""Example selector that selects examples based on SemanticSimilarity."""
+from __future__ import annotations
+
+from itertools import groupby
+from operator import itemgetter
+from typing import Any, Dict, List, Optional, Type
+
+from langchain.embeddings.base import Embeddings
+from langchain.prompts.example_selector.base import BaseExampleSelector
+from langchain.vectorstores.base import VectorStore
+from pydantic import BaseModel, Extra
+
+
+def sorted_values(values: Dict[str, str]) -> List[Any]:
+    """Return a list of values in dict sorted by key."""
+    return [values[val] for val in sorted(values)]
+
+
+class LabelDiversityRandomExampleSelector(BaseExampleSelector, BaseModel):
+    """Example selector that selects examples based on SemanticSimilarity."""
+
+    examples: List[dict]
+    """A list of the examples that the prompt template expects."""
+    k: int = 1
+    """Number of examples to select per label."""
+    label_key: str
+    """Name of the label column/key."""
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+        arbitrary_types_allowed = True
+
+    def add_example(self, example: Dict[str, str]) -> None:
+        self.examples.append(example)
+
+    def select_examples(self, input_variables: Dict[str, str]) -> List[dict]:
+        selected_examples = []
+        sorted_examples = sorted(self.examples, key=itemgetter(self.label_key))
```

Comment:
i didn't quite get this step - if we're grouping by the label next, do we need to do this sorting?

Reply:
groupby just chunks consecutive groups with the same value so it requires the list to be sorted by the value for our use case (otherwise we'd have several chunks of each label interspersed)